### PR TITLE
[IOTDB-4554] Start compaction only when IoTDB is fully recovered

### PR DESCRIPTION
**Pull request by THUMarkLau**
See [IOTDB-4554](https://issues.apache.org/jira/browse/IOTDB-4554).

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/service/DataNode.java [hunk 1]
```diff
@@ -250,6 +252,7 @@ private void active() throws StartupException {
       logger.error("Meet error while starting up.", e);
       throw new StartupException("Error in activating IoTDB DataNode.");
     }
+    activated = true;
```
Comment (yifuzhou) [inline server/src/main/java/org/apache/iotdb/db/service/DataNode.java:255]: It seems that activated cannot turn false when deactivate the datanode.

**Review by yifuzhou — COMMENTED** [review]
Hi, PTAL~

File: server/src/main/java/org/apache/iotdb/db/service/DataNode.java (hunk 1 above)
Comment (THUMarkLau) [reply server/src/main/java/org/apache/iotdb/db/service/DataNode.java:255]: Fixed, thank you.

File: server/src/main/java/org/apache/iotdb/db/service/DataNode.java [hunk 2]
```diff
@@ -316,6 +315,7 @@ private void setUp() throws StartupException, QueryProcessException {
     registerManager.register(RegionMigrateService.getInstance());
 
     registerManager.register(MetricService.getInstance());
+    registerManager.register(CompactionTaskManager.getInstance());
```
Comment (xingtanzjr) [inline server/src/main/java/org/apache/iotdb/db/service/DataNode.java:318]: As we change the sequence of invoking `CompactionTaskManager.getInstance()`, why do we need to add the variable `init` inside the class ?

**Review by xingtanzjr — APPROVED** [review]

**Pull request merged**